### ISSUE TITLE
rubygems installer uses absolute pathnames in generated windows stub batch files

### ISSUE THREAD

**Issue by oxygene** [rubygems/rubygems#2111]
I'd like to propose to use `%~dp0` instead of the hardcoded absolute path to ruby when generating the windows stub batch files.

The rationale is that I'd like to use some gems with jruby in order to deploy a portable self-contained folder that I can put on a build-VM without internet access. Please see https://github.com/jruby/jruby/issues/4890 for my initial bug report that I have wrongfully filed on the jruby project.

Here seems to be the relevant code:
https://github.com/rubygems/rubygems/blob/62dbdc5a06fd8d740e9ce65c8f5619d80366673d/lib/rubygems/installer.rb#L750-L760

**Comment by shevegen** [comment]
Anyone able to test the proposed fix? I currently have no windows platform available.

**Comment by MSP-Greg** [comment]
@shevegen

> Anyone able to test the proposed fix?

Where is the proposed fix?

This has been on my to-do list for a while, and I've got a bookmark to the code (same as above).  Been busy with 2.5 and --enable-frozen-string-literal issues (including in RubyGems)...

I'll get to it, as it creates quite a mess when people (me) try to create portable install sets/images.  I've got code in my build system, but that only affects bin files included in the build.

Within the next few days, Greg
